spec.html

```diff
@@ -4482,6 +4483,22 @@ <h1>Strict Equality Comparison</h1>
   <emu-clause id="sec-operations-on-objects">
     <h1>Operations on Objects</h1>
 
+    <emu-clause id="sec-allocatebasicobject" aoid="AllocateBasicObject">
+      <h1>AllocateBasicObject ( _internalSlotsList_ )</h1>
+      <p>The abstract operation AllocateBasicObject is the source of all ECMAScript objects, including both ordinary objects and exotic objects. It factors out common steps used in creating all objects, and centralizes object creation.</p>
+
+      <emu-alg>
+        1. Let _obj_ be a newly created object with an internal slot for each name in _internalSlotsList_.
+        1. Set _obj_'s essential internal methods to the default ordinary object definitions specified in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>.
+        1. Assert: if the caller will not be overriding both _obj_'s [[GetPrototypeOf]] and [[SetPrototypeOf]] essential internal methods, then _internalsSlotsList_ contains [[Prototype]].
+        1. Assert: if the caller will not be overriding all of _obj_'s [[SetPrototypeOf]], [[IsExtensible]], and [[PreventExtensions]] essential internal methods, then _internalsSlotsList_ contains [[Extensible]].
+        1. If _internalSlotsList_ contains [[Extensible]], then set _obj_.[[Extensible]] to *true*.
+        1. Return _obj_.
+      </emu-alg>
+
+      <p>With regard to exotic objects, the intention is for AllocateBasicObject to be used as a building block for creating specific factory abstract operations (e.g., ArrayCreate or BoundFunctionCreate). In order to encapsulate exotic object creation in such factories, it is required that only the direct caller of AllocateBasicObject modify the return value's essential internal methods.</p>
```

Comment:
Per previous discussions up-thread with Allen, I think stating it as a requirement is better. It's OK to have multiple conformance classes (specification authors vs. implementers) in one document.

Reply:
It can be OK, yes, but:
- Such a document should be clear that it has multiple conformance classes (which I don't think is true of the ES spec); and
- It should be clear which class is addressed by any given requirement (which I don't think is true of the paragraph in question).